insns.def

```diff
@@ -433,12 +433,17 @@ intern
 /* put new array initialized with num values on the stack. */
 DEFINE_INSN
 newarray
-(rb_num_t num)
+(rb_num_t num, VALUE check_kw)
 (...)
 (VALUE val)
 // attr rb_snum_t sp_inc = 1 - (rb_snum_t)num;
 {
-    val = rb_ary_new4(num, STACK_ADDR_FROM_TOP(num));
+    if (check_kw == Qtrue && RHASH_EMPTY_P(*STACK_ADDR_FROM_TOP(1))) {
```

Comment:
Having a branch might slow down `newarray` when it's `check_kw == Qfalse`. Did you consider separating an instruction for `check_kw == Qtrue` case? 

The above suggestion is purely for performance, but also we'd be able to get rid of the unnecessary `check_kw` argument in `opt_newarray_*`.

Reply:
I considered it, and it is probably going to be faster, but wanted to make the minimal changes possible.  I'm definitely fine adding a separate instruction for this if that's what other developers would prefer.